packages/hurl/src/http/response_ip.rs

```diff
@@ -0,0 +1,31 @@
+/*
+ * Hurl (https://hurl.dev)
+ * Copyright (C) 2025 Orange
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *          http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ *
+ */
+use std::str::FromStr;
+
+#[derive(Clone, Debug, Default, Eq, PartialEq)]
+pub struct IpAddr {
+    pub(crate) raw: String,
```

Comment:
Here we don't want someone to have direct access to the `inner` field. 
Instead of `pub(crate) raw: String,`we want `raw: String,`.

The data is encapsulated inside the `IpAddr` and code using `IpAddr` instances doesn't have access to these private fields. If we need, for instance, an access to the `inner` value, we can add methods on `IpAddr`. In this case, we could implement `fmt::Display`so we can call `to_string` on this structure to get the value backed by `inner` field.